x/concentrated-liquidity/lp.go

```diff
@@ -249,7 +274,7 @@ func (k Keeper) updatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 
 	// update position state
 	// TODO: come back to sdk.Int vs sdk.Dec state & truncation
-	err = k.initOrUpdatePosition(ctx, poolId, owner, lowerTick, upperTick, liquidityDelta, joinTime, positionId)
+	err = k.initOrUpdatePosition(ctx, poolId, owner, lowerTick, upperTick, liquidityDelta, joinTime, positionId, noUnderlyingLockId)
```

Comment:
This isn't right, why is this always set to zero? What's the purpose of this parameter?

This is the only place that this method is called

Reply:
You are right, updated here https://github.com/osmosis-labs/osmosis/pull/4882/commits/5855df51844dfb23cf09f734f05ca6d328afb1a1